include/FieldManager.h

```diff
@@ -121,12 +121,12 @@ class FieldManager
   /// field Registry is a static compile-time definition. Care must be taken
   /// not to re-register the same field on the same parts with a conflicting
   /// number of states or components.
-  FieldPointerTypes register_field(
+  FieldPointerTypes register_field (
     const std::string& name,
     const stk::mesh::PartVector& parts,
     const int numStates = 0,
     const int numComponents = 0,
-    const void* init_val = nullptr);
+    const void* init_val = nullptr) const;
```

Comment:
I'm curious how field registry can be a compile-time definition, if it depends on a part-vector which would have run-time contents.

Reply:
I was thinking the type look up could be moved to compile time so that when fields are accessed we wouldn't have to include the type. Registry would still have to be runtime.  However, looking at the dependencies I'm not sure I want to do that anymore. We'd have to expose more of the field information across the code base instead of isolating it to a single `*.C` file.  This could wreak havoc on the compile times. After we convert to the simple fields the type information becomes way less cumbersome for accessing fields.  So I think we should keep it as is.